### ISSUE TITLE
Error during make: invalid conversion from 'const AVCodec*' to 'AVCodec*'

### ISSUE THREAD

**Issue by SCZwangxiao** [dmlc/decord#155]
```bash
root@machine-2824-prod:/home/decord/build# make
Scanning dependencies of target decord
[  2%] Building CXX object CMakeFiles/decord.dir/src/runtime/c_runtime_api.cc.o
[  5%] Building CXX object CMakeFiles/decord.dir/src/runtime/cpu_device_api.cc.o
[  7%] Building CXX object CMakeFiles/decord.dir/src/runtime/dso_module.cc.o
[ 10%] Building CXX object CMakeFiles/decord.dir/src/runtime/file_util.cc.o
[ 13%] Building CXX object CMakeFiles/decord.dir/src/runtime/module.cc.o
[ 15%] Building CXX object CMakeFiles/decord.dir/src/runtime/module_util.cc.o
[ 18%] Building CXX object CMakeFiles/decord.dir/src/runtime/ndarray.cc.o
[ 21%] Building CXX object CMakeFiles/decord.dir/src/runtime/registry.cc.o
[ 23%] Building CXX object CMakeFiles/decord.dir/src/runtime/str_util.cc.o
[ 26%] Building CXX object CMakeFiles/decord.dir/src/runtime/system_lib_module.cc.o
[ 28%] Building CXX object CMakeFiles/decord.dir/src/runtime/thread_pool.cc.o
[ 31%] Building CXX object CMakeFiles/decord.dir/src/runtime/threading_backend.cc.o
[ 34%] Building CXX object CMakeFiles/decord.dir/src/runtime/workspace_pool.cc.o
[ 36%] Building CXX object CMakeFiles/decord.dir/src/video/logging.cc.o
[ 39%] Building CXX object CMakeFiles/decord.dir/src/video/storage_pool.cc.o
[ 42%] Building CXX object CMakeFiles/decord.dir/src/video/video_interface.cc.o
[ 44%] Building CXX object CMakeFiles/decord.dir/src/video/video_loader.cc.o
[ 47%] Building CXX object CMakeFiles/decord.dir/src/video/video_reader.cc.o
/home/decord/src/video/video_reader.cc: In constructor 'decord::VideoReader::VideoReader(std::__cxx11::string, DLContext, int, int, int, int, std::__cxx11::string)':
/home/decord/src/video/video_reader.cc:100:52: error: invalid conversion from 'const AVCodec*' to 'AVCodec*' [-fpermissive]
         AVCodec *local_codec = avcodec_find_decoder(st->codecpar->codec_id);
                                ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/home/decord/src/video/video_reader.cc: In member function 'virtual void decord::VideoReader::SetVideoStream(int)':
/home/decord/src/video/video_reader.cc:149:88: error: invalid conversion from 'AVCodec**' to 'const AVCodec**' [-fpermissive]
     int st_nb = av_find_best_stream(fmt_ctx_.get(), AVMEDIA_TYPE_VIDEO, stream_nb, -1, &dec, 0);
                                                                                        ^~~~
In file included from /home/decord/src/video/ffmpeg/ffmpeg_common.h:24:0,
                 from /home/decord/src/video/threaded_decoder_interface.h:10,
                 from /home/decord/src/video/video_reader.h:10,
                 from /home/decord/src/video/video_reader.cc:7:
/usr/local/include/libavformat/avformat.h:2038:5: note:   initializing argument 5 of 'int av_find_best_stream(AVFormatContext*, AVMediaType, int, int, const AVCodec**, int)'
 int av_find_best_stream(AVFormatContext *ic,
     ^~~~~~~~~~~~~~~~~~~
/home/decord/src/video/video_reader.cc:165:55: error: invalid conversion from 'const AVInputFormat*' to 'AVInputFormat*' [-fpermissive]
             ctx_.device_id, codecpar.get(), fmt_ctx_->iformat));
                                             ~~~~~~~~~~^~~~~~~
In file included from /home/decord/src/video/video_reader.cc:11:0:
/home/decord/src/video/nvcodec/cuda_threaded_decoder.h:49:9: note:   initializing argument 3 of 'decord::cuda::CUThreadedDecoder::CUThreadedDecoder(int, AVCodecParameters*, AVInputFormat*)'
         CUThreadedDecoder(int device_id, AVCodecParameters *codecpar, AVInputFormat *iformat);
         ^~~~~~~~~~~~~~~~~
CMakeFiles/decord.dir/build.make:470: recipe for target 'CMakeFiles/decord.dir/src/video/video_reader.cc.o' failed
make[2]: *** [CMakeFiles/decord.dir/src/video/video_reader.cc.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/decord.dir/all' failed
make[1]: *** [CMakeFiles/decord.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

I'm using: 
- cmake 3.10.2
- make 4.1

**Comment by zhreshold** [comment]
Verified that the problem is caused by the latest ffmpeg, temporary solution is to downgrade libavcodec to ffmpeg<=4.2. I will try to update the code in the future.

**Comment by zhreshold** [comment]
Fixed in #160

**Comment by mangell003** [comment]
When I execute the last command

# make -j3

I get compile error:

In function ‘int stbi_write_bmp_to_func(void (*)(void*, void*, int), void*, int, int, int, const void*)’:
/home/mangell/chiaki-rpi/rpi/include/rpi/stb_image_write.h:514:32: warning: missing initializer for member ‘stbi__write_context::context’ [-Wmissing-field-initializers]
  514 |    stbi__write_context s = { 0 };
